app/errors/ErrorHandler.scala

```diff
@@ -0,0 +1,34 @@
+package errors
+
+import javax.inject.Inject
+import play.api.Logging
+import play.api.http.HttpErrorHandler
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.Results._
+import play.api.mvc.{RequestHeader, Result}
+
+import scala.concurrent.Future
+
+class ErrorHandler @Inject() (val messagesApi: MessagesApi) extends HttpErrorHandler with I18nSupport with Logging {
+
+  override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] = {
+    logger.error(s"Client error with status code $statusCode at path '${request.path}' with message: '$message'")
+
+    Future.successful(
+      Status(statusCode)("A client error occurred: " + message)
+    )
+  }
+
+  override def onServerError(request: RequestHeader, exception: Throwable): Future[Result] = {
+    logger.error(s"Internal server error at path '${request.path}'", exception)
+
+    val response = exception match {
+      case authException: AuthorisationException =>
+        Forbidden(views.html.error(s"Not authorised: ${authException.getMessage}")(request2Messages(request)))
+      case e =>
```

Comment:
Yeah, it's because it's a custom exception. I couldn't find a better way to handle the exception that would let us return a 403 status code _and_ log all the error details without duplicating the error handling in every controller that might hit an authorisation error.

Do you know if there's a more idiomatic way of doing this in Play? I'm mostly trying to remember how other frameworks do things and then trying to find the equivalent in Play, so I'm probably missing things. 🙂

Reply:
I don't know of any other way, this is cleaner than duplicating the logic everywhere so I'm happy.